### ISSUE TITLE
Adjectives and verbs in proper names / verb-headed phrases coerced into nominals

### ISSUE THREAD

**Issue by igorm3** [UniversalDependencies/docs#777]
Hi,

According to the [ud annotation guidelines](https://universaldependencies.org/u/dep/flat.html) `flat` seems to be the most apt relation for handling locales like "Rio de Janeiro", "Sao Paulo", "Sri Lanka", etc., but these consistently get annotated with `compound` relations. Is there a section of the annotation guidelines that is meant to account for this discrepancy or is this a known issue? 

Also, even for English named entities, should "New York" not be tagged with `flat` relations? Thanks in advance for any clarification/guidance on this front.

**Comment by amir-zeldes** [comment]
For common cases you may be able to construct a conversion table from GUM, but there would definitely be a long tail of missed cases. Another approach would be to train a tagger on lower cased text and see if it predicts JJ for certain items with high certainty, then semi-manually disambiguate.

But given the fact that the issue is really broader, I'm not sure that changing to ADJ is the right thing to do - you would need to change to VERB, ADV and others as well, otherwise it's even more inconsistent, no? 

If the objective is just to reproduce amod instead of compound, that seems a bit more realistic. I don't actually find PROPN+amod, or even verbs as PROPN that jarring: in many cases a PROPN acts as part of an NP and you can't really do justice to both analyses. If you have an organization like "Unite the Union" (https://www.unitetheunion.org/) and you say:

  * Unite/PROPN/nsubj the/DET/det Union/PROPN/obj demands action

Then I don't think we would be much better off if we had:

  * Unite/VERB/nsubj ... (or even worse: csubj??)

You have to make a compromise either way, and the PTB view of NNP == lexical part of a name is at least fairly easy to apply. We then still apply the general UD principle of analyzing the internal structure of the name as much as possible in the deprels (so: Union/obj). It's just not possible to do both analyses in one single-tagged tree...

**Comment by nschneid** [comment]
How a phrase acts externally feels like a key part of the morphosyntax, not a mere validator instruction, so I think it would fit in morphological features.

**Comment by Stormur** [comment]
> You can have a phrase acting as a nominal even if the phrase is not headed by a `NOUN`/`PRON`/`PROPN`. This is certainly the case with book/movie titles, and it also holds for sentences where a non-nominal word is the topic, e.g., in: _Unite is not an auxiliary verb in English._
> 
> If the head of the phrase is a verb, then I think it should be tagged `VERB` but attached as `nsubj`, not `csubj` (provided the phrase is a subject). I think the validator will not complain in the subject case but we probably have a problem anyway, because it does check some other dependent positions. The phrase could be an `obj` (not `ccomp`), or an `obl` (not `advcl`), and I'm afraid the validator will currently report `obl` + `VERB` as an error.
> 

This is a very common practice in the newly uploaded Latin treebank, and the validator did not point to any error.

**Comment by amir-zeldes** [comment]
This all sounds great, and I would opt for `CoercedPOS` over `CoercedNominal` in the interest of flexibility, but I have no idea how this could realistically be applied to the English treebanks without substantial manual effort. On top of that, I'm not sure an adjective part of a name (not the head) is really coerced into being a noun: it's just the PTB guidelines which tag it as xpos=NNP, but it is morphosyntactically completely an adjective (and in GUM, also `amod`; in EWT it seems it is usually `compound`)

**Comment by sylvainkahane** [comment]
In our French treebank, we introduced a feature ExtPos (External POS) that we use for idioms, titles but also some others constructions, as soon as we want to specify a different POS towards the governor.
By the way, it was particularly useful when tracking errors.
http://match.grew.fr/?corpus=SUD_French-GSD@latest&custom=6080471c423e4&clustering=N.PhraseType

**Comment by nschneid** [comment]
We have been talking about

* *__Unite__ the Union* demands action — "Unite" is internally a VERB, externally nominal

`CoercedPOS` or `ExtPOS` would allow us to treat some tricky compounds as well, e.g.:

* a stern *don't __mess__ with me* look — "mess" is internally a VERB, externally a noun-modifier (I guess I would call it ADJ)
  - "don't mess with me" in this context could be called a compound (in the morphological sense of the term), and would often be hyphenated signalling its quasi-word status. But clearly its internal structure is of a clause/sentence.

**Comment by amir-zeldes** [comment]
Yes, this would indeed be a good way of encoding cases like "Unite", but those are much more rare - I thought the main issue was adjectives in proper names? For those I think this doesn't work, right?

As for phrasal compound modifiers, since they follow the pattern of noun-noun compounds, I would say "mess" above should have `CoercedPOS=NOUN`. It is probably possible to find the majority of these currently by looking for deprel `compound` which has verb-argstruct-like children (so compound -> nsubj/obj/obl/aux), but this is not a 100% strategy. I find just 6 distinct sentences with constructions like this in GUM, so about once in every 20K tokens: (mostly in reddit, just one in vlog so far)

https://corpling.uis.georgetown.edu/annis/#_q=c190eXBlIF9pXyB0b2sgLT5kZXAgdG9rX2Z1bmM9ImNvbXBvdW5kIiAtPmRlcCB0b2tfZnVuYz0vbnN1YmouKnxvYmp8b2JsfGF1eC8&_c=R1VN&cl=0&cr=0&s=0&l=10

**Comment by nschneid** [comment]
> Yes, this would indeed be a good way of encoding cases like "Unite", but those are much more rare - I thought the main issue was adjectives in proper names? For those I think this doesn't work, right?

We started off talking about ADJ vs. PROPN and got to the question of whether lexical words in names need to be tagged as PROPN because sometimes they can be the head of the name and function nominally with respect to the rest of the sentence.

It seems like there is a tentative consensus that a feature could be a workaround in the latter case, allowing us to stick to VERB for "Unite". This is reassuring for the prospect of a general policy that PROPN should be limited to lexical nominals, and not cover adjectives/verbs (in an ideal UD world).

Whether it is worth the effort to convert the non-nouns that are currently PROPN is another question (particularly in EWT which doesn't use amod for the adjectives in proper names). But if we can make it work practically I think it's the best analysis.

> As for phrasal compound modifiers, since they follow the pattern of noun-noun compounds, I would say "mess" above should have `CoercedPOS=NOUN`. It is probably possible to find the majority of these currently by looking for deprel `compound` which has verb-argstruct-like children (so compound -> nsubj/obj/obl/aux), but this is not a 100% strategy. I find just 6 distinct sentences with constructions like this in GUM, so about once in every 20K tokens: (mostly in reddit, just one in vlog so far)
> 
> https://corpling.uis.georgetown.edu/annis/#_q=c190eXBlIF9pXyB0b2sgLT5kZXAgdG9rX2Z1bmM9ImNvbXBvdW5kIiAtPmRlcCB0b2tfZnVuYz0vbnN1YmouKnxvYmp8b2JsfGF1eC8&_c=R1VN&cl=0&cr=0&s=0&l=10

How to distinguish noun-noun and adjective-noun modification when the modifier is headed neither by an adjective nor a noun?

My gut feeling of ADJ is probably because adjectives occur in fewer syntactic contexts than nouns, and attributive modification is the prototypical context of adjectives. But I could live with calling these nominal phrases.

**Comment by dan-zeman** [comment]
> > If the head of the phrase is a verb, then I think it should be tagged `VERB` but attached as `nsubj`, not `csubj` (provided the phrase is a subject). I think the validator will not complain in the subject case but we probably have a problem anyway, because it does check some other dependent positions. The phrase could be an `obj` (not `ccomp`), or an `obl` (not `advcl`), and I'm afraid the validator will currently report `obl` + `VERB` as an error.
> 
> This is a very common practice in the newly uploaded Latin treebank, and the validator did not point to any error.

OK. This is good in the short term. But if we arrive at one of the feature proposals as a consensual recommendation, then the validator could actually look for new types of errors. Maybe `ExtPos` should be the winner if it is already used in the French treebanks. (Now that @sylvainkahane mentioned it, I recall that it has been discussed in one of the issues here.)

**Comment by Stormur** [comment]
I don't think it should count as an error. I actually don't think such a behaviour warrants any feature, because we already have two layers of annotation, POS and deprels, which together can give notice of "deviant" cases like these. But maybe it could be possible to introduce permitted combinations of POS/deprels like it is for morpholexical features now.

**Comment by amir-zeldes** [comment]
> My gut feeling of ADJ is probably because adjectives occur in fewer syntactic contexts than nouns, and attributive modification is the prototypical context of adjectives. But I could live with calling these nominal phrases.

Well, I guess my gut feeling is NOUN, but probably mainly because I'm used to thinking of compound components as being nouns due to German, Hebrew and Arabic. I guess English, like German, has some proper A+N compounds, so things like blackbird and White House (based on the Compound Stress Rule), but I think the vast majority are N+N, so I would just assume that is the default conversion when something is a compound modifier. It's really hard to imagine a diagnostic context - I guess phrasal modifiers are usually not comparable and negatable with un-? But under extreme conditions maybe you could find an example like that...

In any case, it's a very rare phenomenon, so we can decide whatever we like and apply it by hand. The much bigger issue is English modifier adjectives which are part of names. As long as xpos stays NNP for compatibility I don't really care if they're ADJ or PROPN, and I'm willing to change upos to ADJ in GUM since it's easy enough. But since it would not be easy for EWT, I'm not sure if it's a good idea, since it will only make the two corpora more dissimilar. 

If someone has the energy to disambiguate, I'm willing to train a sequence tagger on GUM deprels and produce a predicted version of whether each NNP in EWT is an adjective, but it won't be 100% accurate of course.

**Comment by arademaker** [comment]
@sylvainkahane I saw from your example above that you also use `idiom=` in the MISC field, right? How do you annotate idioms? If you add the annotation in the head of the phrase, how to deal with dependents that are not part of the expression?

**Comment by sylvainkahane** [comment]
@arademaker In SUD we do not use the `fixed` relation, but add an internal syntactic structure as soon as it is possible and use the link `unk` (unknown) when we cannot recognize any syntactic construction. The idiom itself is encoded by features on the nodes: `Idiom=Yes` on the head and `InIdiom=Yes `on other nodes. When converted into UD, the SUD internal structure of idioms is lost and `fixed` relations are used. The features `Idiom` and `InIdiom` become useless.
